atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -202,6 +208,29 @@ public static SerializableTransactionManager create(
         return transactionManager;
     }
 
+    private static PersistentLockService createAndRegisterPersistentLockService(KeyValueService kvs, Environment env) {
+        if (!kvs.supportsCheckAndSet()) {
+            return new NoOpPersistentLockService();
+        }
+
+        PersistentLockService pls = StandardPersistentLockService.create(kvs);
+        env.register(pls);
+        env.register(new CheckAndSetExceptionMapper());
+        return pls;
+    }
+
+    private static KeyValueService getKeyValueService(ServiceDiscoveringAtlasSupplier atlasFactory,
+            LockAndTimestampServices lts) {
+        KeyValueService rawKvs = atlasFactory.getKeyValueService();
+        KeyValueService kvs = NamespacedKeyValueServices.wrapWithStaticNamespaceMappingKvs(rawKvs);
+        kvs = ValidatingQueryRewritingKeyValueService.create(kvs);
+        kvs = ProfilingKeyValueService.create(kvs);
+        kvs = SweepStatsKeyValueService.create(kvs, lts.time());
+
+        TransactionTables.createTables(kvs);
```

Comment:
It was done previously as part of the main .create method (line 141 in the old code). I pulled out a method that grabs the KVS.

This ensures that we don't give back a KVS that hasn't been set up, so maybe `getInitialisedKeyValueService` would be a better name?